go/vt/servenv/servenv.go

```diff
@@ -133,9 +133,9 @@ func Init() {
 
 	// Once you run as root, you pretty much destroy the chances of a
 	// non-privileged user starting the program correctly.
-	if uid := os.Getuid(); uid == 0 {
-		log.Exitf("servenv.Init: running this as root makes no sense")
-	}
+	// if uid := os.Getuid(); uid == 0 {
```

Comment:
I had to do this initially as we were just getting started with how to run vtgateproxy and I needed it to execute on root to be able to access the files.

@dedelala @rjlaine @henryr with the more recent docker packaging, can we remove the need for this to run as root? 

If so then we don't need this hack

Reply:
As long as the gRPC credentials have `root:root 0600`, we'll need to run the proxy as root. We could add a group to the host which has read access to certain things, eg. `vitess`, and then change file permissions to `root:vitess 0640`, and run the container with `--group-add "$vitess_gid"`

Reply:
Makes sense to me. Either way we wouldn't aim to upstream this particular part, even if we choose not to do this and just keep the commented-out code in Slack's fork.